readthedocs/config/config.py

```diff
@@ -148,7 +163,7 @@ class BuildConfigBase(object):
 
     def __init__(self, env_config, raw_config, source_file):
         self.env_config = env_config
-        self.raw_config = raw_config
+        self.raw_config = copy.deepcopy(raw_config)
```

Comment:
By doing this, we are not solving my comment from https://github.com/rtfd/readthedocs.org/pull/4740#discussion_r238320993 I'd say.

> I think it's not a good idea to change the raw_config object. By doing this, it won't be "raw" anymore and it will bring potential confusions.

If we copy it, we don't touch the original one, but the meaning of the variable is still confusing to me.

Reply:
I guess we can make it private

Reply:
https://github.com/rtfd/readthedocs.org/issues/4966